Tangem/Modules/Markets/TokenList/MarketsViewModel.swift

```diff
@@ -509,18 +509,53 @@ private extension MarketsListDataProvider.Event {
 }
 
 import Moya
+import Alamofire
 
 extension Error {
     var marketsAnalyticsParams: [Analytics.ParameterKey: String] {
         var analyticsParams = [Analytics.ParameterKey: String]()
-        if let error = self as? LocalizedError {
-            analyticsParams[.errorDescription] = error.localizedDescription
+
+        analyticsParams[.errorMessage] = (self as? LocalizedError)?.localizedDescription
+        analyticsParams[.errorCode] = marketsErrorCode(for: self)
+        analyticsParams[.errorType] = marketsErrorType(for: self).rawValue
+
+        return analyticsParams
+    }
+
+    private func marketsErrorCode(for: Error) -> String {
+        if let moyaError = self as? MoyaError,
+           case .statusCode(let response) = moyaError {
+            return String(response.statusCode)
+        } else {
+            return Analytics.ParameterValue.marketsErrorCodeIsNotHTTPError.rawValue
         }
+    }
+
+    private func marketsErrorType(for error: Error) -> Analytics.ParameterValue {
+        switch self {
+        case let moyaError as MoyaError:
+            switch moyaError {
+            case .statusCode:
```

Comment:
А так не получится ? Что бы вложенности не было
```suggestion
        case MoyaError.statusCode(let code):
```

Reply:
не сработало: Pattern of type 'MoyaError' cannot match 'Self'. 
поправил немного по-другому